server/src/main/scala/org/scassandra/server/actors/BatchHandler.scala

```diff
@@ -83,15 +84,19 @@ class BatchHandler(activityLog: ActivityLog,
       processBatch(header, batch, batchQueries, connection)
   }
 
-  def processBatch(header: FrameHeader, batch: Batch, batchQueries: Seq[BatchQuery], recipient: ActorRef) = {
+  def processBatch(header: FrameHeader, batch: Batch, batchQueries: Seq[BatchQuery], recipient: ActorRef): Unit = {
     val execution = BatchExecution(batchQueries, batch.consistency, batch.serialConsistency, batch.batchType, batch.timestamp)
-    activityLog.recordBatchExecution(execution)
-    val prime = batchPrimeStore(execution)
-    prime.foreach(p => log.info("Found prime {} for batch execution {}", p, execution))
-    writePrime(batch, prime, header, recipient=Some(recipient), alternative = Some(Reply(VoidResult)), consistency = Some(batch.consistency))
+    activityLog ! RecordBatch(execution)
```

Comment:
One subtle difference is that since `!` is used here, we now can't guarantee that the execution is added to the ActivityLog before exiting `processBatch`.  On the other hand, it's probably incredibly unlikely that will be an issue (since there's more work to finish work and send response back to client, and the client has to read activity log).  If wanted to be strict about that we could use `?` here and defer on response to know that the activity log has been updated.  This would definitively prevent situations where the client gets a response before activity log is updated and goes to get it and it's not there, however I also think that may be very hard to reproduce, so maybe we don't need to worry about it.

Reply:
Yeah good spot. I had to do this in all the routes for the same reason.